### PR TITLE
cooler cooler explosions now using particles

### PR DESCRIPTION
**Pull request by tyeagg**
## About The Pull Request
directed by michael bay

light and heavy explosion

https://github.com/tgstation/TerraGov-Marine-Corps/assets/98577688/d79aaea9-9ced-43ad-8a23-807528be768e

devastation explosions and lots of light explosions at once

https://github.com/tgstation/TerraGov-Marine-Corps/assets/98577688/802f8f1d-898f-4df0-9bfe-e23ff2b93357
## Why It's Good For The Game
looks cool
## Changelog
:cl:
add: particle explosions
del: effect system for explosions
imageadd: more smoke icons
code: removed the small anim arg from explosions
/:cl:

### CODE REVIEW

**Comment by Kapu1178** [comment]
"PRIVATE, DO YOU SEE THAT BUILDING?"
"YES SIR"
"WELL I DON'T WANT TO!"
image

**Review by Lumipharon — APPROVED** [review]

File: code/game/objects/effects/temporary_visuals/explosions.dm [hunk 1]
```diff
@@ -0,0 +1,315 @@
...
+	addtimer(VARSET_CALLBACK(explosion_smoke.particles, count, 0), 5)
+	QDEL_IN(explosion_smoke, 25)
+	addtimer(VARSET_CALLBACK(dirt_kickup.particles, count, 0), 10)
+	QDEL_IN(dirt_kickup, 25)
+	addtimer(VARSET_CALLBACK(falling_debris.particles, count, 0), 5)
+	QDEL_IN(falling_debris, 25)
+	addtimer(VARSET_CALLBACK(sparks.particles, count, 0), 5)
+	QDEL_IN(sparks, 25)
+	addtimer(VARSET_CALLBACK(large_kickup.particles, count, 0), 5)
+	QDEL_IN(large_kickup, 25)
```
Comment (TiviPlus) [inline code/game/objects/effects/temporary_visuals/explosions.dm:315]: You can just make this 2 timers and avoid a lot of timer churn

File: code/game/objects/effects/temporary_visuals/explosions.dm [hunk 2]
```diff
@@ -0,0 +1,315 @@
...
+	generate_particles(radius, small, large)
+	var/image/I = image(icon, src, icon_state, 10, -32, -32)
+	if(iswater(get_turf(src)))
+		icon_state = null
+		return
+	var/matrix/rotate = matrix()
+	rotate.Turn(rand(0, 359))
+	I.transform = rotate
+	overlays += I //we use an overlay so the explosion and light source are both in the correct location plus so the particles don't rotate with the explosion
+	icon_state = null
```
Comment (TiviPlus) [inline code/game/objects/effects/temporary_visuals/explosions.dm:258]: You can move I to below the return

**Review by TiviPlus — COMMENTED** [review]
Rest looks good enough

File: code/game/objects/effects/temporary_visuals/explosions.dm [hunk 3]
```diff
@@ -0,0 +1,314 @@
...
+	///explosion smoke particle holder
+	var/obj/effect/abstract/particle_holder/explosion_smoke
+	///debris dirt kickup particle holder
+	var/obj/effect/abstract/particle_holder/dirt_kickup
+	///falling debris particle holder
+	var/obj/effect/abstract/particle_holder/falling_debris
+	///sparks particle holder
+	var/obj/effect/abstract/particle_holder/sparks
+	///large dirt kickup particle holder
+	var/obj/effect/abstract/particle_holder/large_kickup
```
Comment (TiviPlus) [inline code/game/objects/effects/temporary_visuals/explosions.dm:244]: Not nulled in destroy, leaving hanging ref

**Review by TiviPlus — CHANGES_REQUESTED** [review]

**Review by Lumipharon — APPROVED** [review]

**Pull request merged**